marlowe-actus/src/Language/Marlowe/ACTUS/Analysis.hs

```diff
@@ -23,6 +24,18 @@ import           Prelude                                               hiding (F
 genProjectedCashflows :: ContractTerms -> [CashFlow]
 genProjectedCashflows = sampleCashflows (const $ RiskFactors 1.0 1.0 1.0 0.0)
 
+postProcessSchedule :: ContractTerms -> [(EventType, ShiftedDay)] -> [(EventType, ShiftedDay)]
+postProcessSchedule ct s =
+    let trim = L.dropWhile (\(_, d) -> calculationDay d < ct_SD ct)
+        prioritised = [AD, IED, PR, PI, PRF, PY, FP, PRD, TD, IP, IPCI, IPCB, RR, PP, CE, MD, RRF, SC, STD, DV, XD, MR]
+        priority :: (EventType, ShiftedDay) -> Integer
+        priority (event, _) = fromJust $ M.lookup event $ M.fromList (zip prioritised [1..])
+        simillarity (_, l) (_, r) = calculationDay l == calculationDay r
+        regroup = L.groupBy simillarity
+        overwrite = map (L.head . sortOn priority) . regroup
+    in (overwrite . trim) s
```

Comment:
It would be possible ETA reduce this, since it uses `.`.
I think it would also be possible to write:
```haskell
overwrite $ trim s
```

Reply:
fixed - I made last expression point-free